python/lsst/pipe/base/task.py

```diff
@@ -111,7 +111,7 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
             if config is None:
                 config = getattr(parentTask.config, name)
             self._taskDict = parentTask._taskDict
-            self.log = Log.getLogger(parentTask.log.getName() + '.' + name)
+            loggerName = parentTask.log.getName() + '.' + name
```

Comment:
Please don't change the original logic unless you have a compelling reason to do so (and if you do have such a reason, please explain it in the commit message). This changes the API and I think the old one sufficed, though it was not properly documented (thus my small commit which updated the documentation for the log argument, which your change then rendered incorrect).

I encourage you to compare your code to the master branch. I think you'll find that almost no changes are required other than documenting the new type of the log argument and tweaking the way the default log is obtained.